src/app/domain/geolocation/helpers/index.ts

```diff
@@ -4,26 +4,26 @@ import Minilog from 'cozy-minilog'
 
 const log = Minilog('📍 Geolocation')
 
-export const getTs = location => {
+export const getTs = (location: { timestamp: string }): number => {
   return parseISOString(location.timestamp).getTime() / 1000
 }
 
-export const parseISOString = ISOString => {
-  let b = ISOString.split(/\D+/)
+export const parseISOString = (ISOString: string): Date => {
+  const b = ISOString.split(/\D+/).map(Number)
   return new Date(Date.UTC(b[0], --b[1], b[2], b[3], b[4], b[5], b[6]))
 }
```

Comment:
@paultranvan is it okay for you?

Reply:
I'm not sure what it means for [CozyGPSMemory](https://github.com/cozy/CozyGPSMemory): will it require some extra ts config? 
I'm a bit worried that it may introduce more harm than good

Reply:
> I'm not sure what it means for [CozyGPSMemory](https://github.com/cozy/CozyGPSMemory): will it require some extra ts config? I'm a bit worried that it may introduce more harm than good

I don't think it should impact CozyGPSMemory, I think the issue here could be an unexpected behaviour at runtime with the casting to Number explicitly (this was required for stronger type inference).

Reply:
> I think the issue here could be an unexpected behaviour at runtime with the casting to Number explicitly

I'm not sure what you imply here: do you mean an issue could arise if the variable is actually not a number (which is normal), or that this issue could arise anyway?